### DISCUSSION TITLE
Cannot add my SearXNG instance to Firefox search engine list

### DISCUSSION THREAD

**Discussion by rickysixx** [searxng/searxng#2549] [Q&A] [answered]
I am using Firefox on Linux and I get an error when I try to add my private SearXNG instance to Firefox.

I am following [Firefox's guide](https://support.mozilla.org/en-US/kb/add-or-remove-search-engine-firefox#w_add-a-search-engine-from-the-address-bar) to add a search engine from the address bar:

![image](https://github.com/searxng/searxng/assets/51907568/815f47fc-ecf6-4685-939b-d7faa019ca3a)

When I click on `Add "searxng"` I get this error:

![image](https://github.com/searxng/searxng/assets/51907568/899f7455-aeea-44ee-89f4-5ecb90f99964)

I have tried to access the same URL with `curl` and I correctly get the Opensearch XML file:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
  <ShortName>searxng</ShortName>
  <LongName>SearXNG metasearch</LongName>
  <Description>SearXNG is a metasearch engine that respects your privacy.</Description>
  <InputEncoding>UTF-8</InputEncoding>
  <Image type="image/png">http://searxng.home.arpa/static/themes/simple/img/favicon.png</Image>
  <Url rel="results" type="text/html" method="GET" template="http://searxng.home.arpa/search?q={searchTerms}"/>
    <Url rel="suggestions" type="application/x-suggestions+json" method="GET" template="http://searxng.home.arpa/autocompleter?q={searchTerms}"/>
  <Url rel="self" type="application/opensearchdescription+xml"  method="GET" template="http://searxng.home.arpa/opensearch.xml?method=GET&amp;autocomplete=google" />
  <Query role="example" searchTerms="SearXNG" />
  <moz:SearchForm>http://searxng.home.arpa/search</moz:SearchForm>
</OpenSearchDescription>
```

Any suggestion on how to fix this?

**Comment by return42** [comment, marked as answer]
Hm, strange .. might be related to any http / https restrictions of the browser? What happens when you open the URL directly from the URL bar of the browser?

**Comment by rickysixx** [comment]
> Hm, strange .. might be related to any http / https restrictions of the browser? What happens when you open the URL directly from the URL bar of the browser?

You were right. I had HTTPS everywhere enabled in Firefox settings. When I disabled it I was able to add the search engine without errors.

That was strange, because if I try to navigate the URL with Firefox (with HTTPS everywhere enabled), it asks me to download the XML file (which I see is a valid Opensearch XML file, if I open it with Firefox) and it does not complain about the missing HTTPS.

Anyway, problem solved. Thank you for your answer!
